Deploy/stacks/dynamic/common-scripts/common_functions.sh

```diff
@@ -6,12 +6,57 @@ set -e
 get_executables(){
     if [ -z "$EXECUTABLE" ]; then
         # Ensure compatibility with podman
-        if command -v podman &> /dev/null
-        then
+        if command -v podman &> /dev/null; then
             #>&2 echo "ERROR Podman cannot run in swarm mode so will have to move to Kubenetes before this actually works"
             # The "--podman-build-args" argument requires podman compose version 0.1.8
-            pip3 install --user -q 'podman-compose==1.0.3'
-            
+
+            if [ -n "$VIRTUAL_ENV" ]; then
+                #echo "Info: Using Python virtual environment $VIRTUAL_ENV..."
+
+                pip_executable="pip3"
+                if command -v "$pip_executable" &> /dev/null; then
+
+                    package_name="podman-compose"
+                    desired_version="1.0.3"
+                    # Check if the package is installed
+                    if $pip_executable show "$package_name" >/dev/null 2>&1; then
+                        installed_version=$($pip_executable show "$package_name" | grep Version | awk '{print $2}')
+                        # Compare installed version with desired version
+                        if [[ "$installed_version" != "$desired_version" ]]; then
+                            echo "ERROR: Installed version of $package_name is $installed_version, but the desired version is $desired_version. Please either remove the incompatible version or (better) use a different virtual environment."
+                            exit 1
+                        else
+                            #echo "Info: Found $package_name version $desired_version."
+                            :
+                        fi
+                    else
+                        echo "Info: $package_name not found. Installing version $desired_version..."
+                        $pip_executable install -q "$package_name==$desired_version"
+                    fi
+
+                    # Patch podman-compose
+                    patch_executable="patch"
+                    if command -v "$patch_executable" &> /dev/null; then
+                        sitepackage_path=$(python3 -c 'import sysconfig; print(sysconfig.get_paths()["purelib"])')
+                        filename="$sitepackage_path/podman_compose.py"
+                        # NB The -N argument ensures the patch is applied only in forward direction, i.e. reverse or re-application will be ignored. This makes it idempotent!
+                        $patch_executable -u -N $filename -i "${SCRIPTS_DIR}/podman/podman_compose_v$desired_version.patch" > /dev/null 2>&1 || true
+                    else
+                        echo "ERROR: '$patch_executable' command not found. Please install it."
```

Comment:
The `patch` command is not installed on all distributions by default. Changing to use `git apply` should be more compatible.
The following changes seem to work:
```suggestion
                    patch_executable=("git" "apply")
                    if command -v "${patch_executable[@]}" &> /dev/null; then
                        sitepackage_path=$(python3 -c 'import sysconfig; print(sysconfig.get_paths()["purelib"])')
                        filename="$sitepackage_path/podman_compose.py"
                        # NB The -N argument ensures the patch is applied only in forward direction, i.e. reverse or re-application will be ignored. This makes it idempotent!
                        "${patch_executable[@]}" -u -N $filename -i "${SCRIPTS_DIR}/podman/podman_compose_v$desired_version.patch" > /dev/null 2>&1 || true
                    else
                        echo "ERROR: '${patch_executable[*]}' command not found. Please install it."
```

Reply:
I've updated the command so that it works, this needs to be batch applied with a change to the patch file.